torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -914,20 +914,30 @@ def _test_all_reduce_helper(
             expected_value,
             cuda=False,
             rank_to_GPU=None,
+            async_op=False,
         ):
             for src in group:
                 if rank == src:
                     tensor = _build_tensor(src + 1).fill_(master_value)
-                    if cuda:
-                        tensor = tensor.cuda(rank_to_GPU[rank][0])
-                    dist.all_reduce(tensor, op, group_id)
-                    self.assertEqual(tensor, _build_tensor(src + 1, expected_value))
                 else:
                     tensor = _build_tensor(src + 1).fill_(worker_value)
-                    if cuda:
-                        tensor = tensor.cuda(rank_to_GPU[rank][0])
-                    dist.all_reduce(tensor, op, group_id)
-                    self.assertEqual(tensor, _build_tensor(src + 1, expected_value))
+                if cuda:
+                    tensor = tensor.cuda(rank_to_GPU[rank][0])
+                with torch.autograd.profiler.profile() as prof:
+                    work = dist.all_reduce(tensor, op, group_id, async_op=async_op)
+                    if async_op:
+                        work.wait()
+                        work._get_profiling_future().wait()
```

Comment:
What is the reason to have these 3 lines? Shouldn't the profiling work even without this?